### PR TITLE
[HTP-0001] Facebook Adapter: ORTB Bidding Endpoint Change

### PR DESCRIPTION
**Pull request by sanyamgarg93**
## Type of Change

- [ ] New adapter
- [x] Feature Update
- [ ] Bug Fix
- [ ] Code Refactor
- [ ] Other

### Related Issue

#

### CODE REVIEW

File: facebook/facebook-htb.js [hunk 1]
```diff
@@ -0,0 +1,680 @@
...
+    /**
+     * Returns true or false depending on whether the requested format
+     * can be considered as banner format under ORTB guidelines
+     *
+     * @param {object} parcel
+     *
+     * @return {boolean}
+     */
+    function __isBannerFormat(parcel) {
+        var isFullwidth = parcel.xSlotRef.adFormat && parcel.xSlotRef.adFormat === 'fullwidth';
```
Comment (sanyamgarg93) [inline facebook/facebook-htb.js]: Check again

File: facebook/facebook-htb.js [hunk 2]
```diff
@@ -0,0 +1,680 @@
...
+        var dnt = Browser.topWindow.navigator.doNotTrack === 'yes'
+                || Browser.topWindow.navigator.doNotTrack === '1'
+                || Browser.topWindow.navigator.msDoNotTrack === '1' ? 1 : 0;
+        var platformVersion = SpaceCamp.version;
+
+        var queryObj = {
+            id: auctionIdentifierId,
+            imp: [],
+            site: {
+                page: pageUrl,
```
Comment (sanyamgarg93) [inline facebook/facebook-htb.js]: How is url being passed in older version?

File: facebook/facebook-htb.js [hunk 3]
```diff
@@ -0,0 +1,680 @@
...
+                 * This section maps internal returnParcels and demand returned from the bid request.
+                 * In order to match them correctly, they must be matched via some criteria. This
+                 * is usually some sort of placements or inventory codes. Please replace the someCriteria
+                 * key to a key that represents the placement in the configuration and in the bid responses.
+                 */
+
+                /* ----------- Fill this out to find a matching bid for the current parcel ------------- */
+                if (curReturnParcel.htSlot.getId() === bids[i].impid) {
+                    curBid = bids[i];
+                    bids.splice(i, 1);
```
Comment (sanyamgarg93) [inline facebook/facebook-htb.js]: check again

File: facebook/facebook-htb.js [hunk 4]
```diff
@@ -0,0 +1,680 @@
...
+            var placementId = String(parcel.xSlotRef.placementId);
+            var appId = String(placementId.split('_')[0]);
+
+            if (!__appIdDict[appId]) {
+                __appIdDict[appId] = 1;
+            } else {
+                __appIdDict[appId]++;
+            }
+
+            if (__seenPlacementIds[placementId]) {
```
Comment (sanyamgarg93) [inline facebook/facebook-htb.js]: Repeated placement IDs are allowed

File: facebook/facebook-htb.js (hunk 1 above)
Comment (sanyamgarg93) [reply facebook/facebook-htb.js]: Check edge cases for Native and fullwidth

File: facebook/facebook-htb.js (hunk 4 above)
Comment (sanyamgarg93) [reply facebook/facebook-htb.js]: Test with same placement IDs

File: facebook/facebook-htb.js (hunk 2 above)
Comment (sanyamgarg93) [reply facebook/facebook-htb.js]: Is there a better way to generate dnt and pageurl?

File: facebook/facebook-htb.js (hunk 1 above)
Comment (sanyamgarg93) [reply facebook/facebook-htb.js]: Passing incorrect slot info should return error in ORTB. Try passing a mRect placement as native and viceversa.
